nifi-extension-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/pcap/SplitPCAP.java

```diff
@@ -0,0 +1,248 @@
+
+                            byte[] packetHeaderArray = new byte[Packet.PACKET_HEADER_LENGTH];
+                            bInStream.read(packetHeaderArray, 0, Packet.PACKET_HEADER_LENGTH);
+                            Packet currentPacket = new Packet(packetHeaderArray, templatePcap);
+
+                            if (currentPacket.totalLength() > pcapMaxSize) {
+                                throw new IOException("PCAP contains a packet larger than the max size.");
+                            }
+
+                            if (currentPacket.isInvalid()) {
+                                throw new IOException("PCAP contains an invalid packet.");
+                            }
+
+                            byte[] packetBodyArray = new byte[(int) currentPacket.expectedLength()];
+
+                            bInStream.read(packetBodyArray, 0, (int) currentPacket.expectedLength());
+                            currentPacket.setBody(packetBodyArray);
+
+                            if (currentPcapTotalLength + currentPacket.totalLength() > pcapMaxSize) {
+
+                                templatePcap.packets().addAll(loadedPackets);
+                                FlowFile newFlowFile = session.create(flowFile);
+                                try (final OutputStream out = session.write(newFlowFile)) {
+                                    out.write(templatePcap.readBytesFull());
+                                    splitFilesList.add(newFlowFile);
+
+                                } catch (IOException e) {
+                                    throw new IOException(e.getMessage());
+                                }
+
+                                loadedPackets.clear();
+                                loadedPackets.add(currentPacket);
+                                currentPcapTotalLength = PCAP.PCAP_HEADER_LENGTH;
+                                templatePcap.packets().clear();
+                            }
+
+                            loadedPackets.add(currentPacket);
+                            currentPcapTotalLength += currentPacket.totalLength();
+                        }
+
+                        // If there are any packets left over, create a new flowfile.
+                        if (!loadedPackets.isEmpty()) {
+                            templatePcap.packets().addAll(loadedPackets);
+                            FlowFile newFlowFile = session.create(flowFile);
+                            try (final OutputStream out = session.write(newFlowFile)) {
+                                out.write(templatePcap.readBytesFull());
+                                splitFilesList.add(newFlowFile);
+
+                            } catch (IOException e) {
+                                throw new IOException(e.getMessage());
+                            }
+                        }
+
+                    } catch (IOException e) {
+                        for (FlowFile splitFile : splitFilesList) {
```

Comment:
@JackHintonSmartDCSIT I would take a look at `SplitRecord` as it has exception handling in the callback and and around the `session.read` but it handles all `session.remove` in the catch block for `session.read`.